data/campaigns/World_Conquest/lua/game_mechanics/promote_commander.lua

```diff
@@ -19,8 +19,11 @@ on_event("die", function(cx)
 	}
 	commander = commander[1]
 	if commander then
+		commander:extract()
+		commander.id = "wc_leader"..u.side
```

Comment:
Wouldn't it need to be `wc2_leader` to match the save_id? If you use the id of the previous leader (u.id) instead it doesn't matter anyway.